### PR TITLE
Closes #353 update release strategy

### LINKED ISSUE #353: Documentation: Update "Release Strategy"
### Please select a category the issue is focused on?

Developer Guides

### Let us know where something needs a refresh or put your idea here!

Update "Release Strategy" with respect to the new branching strategy.

### PR DESCRIPTION
**Pull request by bundfussr**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiraldev/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by bundfussr** [comment]
@ddsjoberg , I have tried to update the steps required for a release. Please have a look and feel free to update if I got something wrong.

- I am not sure regarding the website updates. I hope a release triggers a workflow which creates a new `vx.y.z` folder in `gh-pages` and stores the documentation of the released versions there. Is this correct?
- There were some discussions regarding using `Remotes:` in `DESCRIPTION` and specifying development versions as minimum version in `DESCRIPTION`. I am not sure if we want to do this and in the affirmative when we should do it. Could you clarify?
If we use it, I assume we need to add something that these are removed before sending the package to CRAN. Do you agree?

File: vignettes/release_strategy.Rmd [hunk 1]
```diff
@@ -39,34 +39,58 @@ updates via the community meetings, especially any anticipated breaking changes.
...
+    a) Create a new feature branch from `main`.
+    a) Update the `Versions:` field in the `DESCRIPTION` file.
+    a) Update `NEWS.md` (replace "(development version)" with the version to be
+    released)
+    a) Create a PR into `main` and merge it. The title of the PR must contain "[skip vbump]".
+    Otherwise, the vbump workflow sets the version to a development version and
+    CRAN will reject the submission.
+1) Bundle up (from `main`) and send off to CRAN. See the [chapter](https://r-pkgs.org/release.html#decide-the-release-type) in R Packages for more details.
 1) If CRAN asks for modifications, repeat steps 1-2 as necessary.
 1) Once the package is accepted and available on CRAN, a [GitHub action](https://github.com/pharmaverse/admiral/actions/workflows/pages/pages-build-deployment) is set up to rebuild the `{admiral}` website with all the updates for this release.
```
Comment (bundfussr) [inline vignettes/release_strategy.Rmd:56]: @ddsjoberg , what is meant by "is set up"? Is this done automatically or does it require an action from the developer?

File: vignettes/release_strategy.Rmd [hunk 2]
```diff
@@ -39,34 +39,58 @@ updates via the community meetings, especially any anticipated breaking changes.
...
-
-1) Verify that all CI/CD checks and then bundle up and send off to CRAN. See the [chapter](https://r-pkgs.org/release.html#decide-the-release-type) in R Packages for more details.
+## Planned Release
+
+A package release is done in the following steps:
+
+1) Create the release version:
+    a) Create a new feature branch from `main`.
+    a) Update the `Versions:` field in the `DESCRIPTION` file.
+    a) Update `NEWS.md` (replace "(development version)" with the version to be
```
Comment (ddsjoberg) [inline vignettes/release_strategy.Rmd:49]: These two steps can be done with a call to `usethis::use_version()`. You'll be prompted whether you're making a major, minor, or patch version bump.

File: vignettes/release_strategy.Rmd [hunk 3]
```diff
@@ -39,34 +39,58 @@ updates via the community meetings, especially any anticipated breaking changes.
 
 # Package Release Process
 
-## Quarterly Release
-
-A package release is done in four parts:
-
-1) Verify that all CI/CD checks and then bundle up and send off to CRAN. See the [chapter](https://r-pkgs.org/release.html#decide-the-release-type) in R Packages for more details.
+## Planned Release
```
Comment (ddsjoberg) [inline vignettes/release_strategy.Rmd:42]: There are many other steps we should be checking before a release that are enumerated in the GitHub Release Issue created with `usethis::use_release_issue()`. This includes running R CMD Check on the CRAN Windows devel env, running the CRAN-compliant URL checker (we use the lychee too, which has some different requirements from CRAN), updating CRAN comments, polishing the NEWS file, running reverse dependency checks, etc.

File: vignettes/release_strategy.Rmd (hunk 1 above)
Comment (ddsjoberg) [reply vignettes/release_strategy.Rmd:56]: I don't think there is anything new that is set up. Whenever we make an admiral release, the admiral website is built. I am not aware of anything else occuring.

**Comment by ddsjoberg** [comment]
> * I am not sure regarding the website updates. I hope a release triggers a workflow which creates a new `vx.y.z` folder in `gh-pages` and stores the documentation of the released versions there. Is this correct?

You are correct!

> * There were some discussions regarding using `Remotes:` in `DESCRIPTION` and specifying development versions as minimum version in `DESCRIPTION`. I am not sure if we want to do this and in the affirmative when we should do it. Could you clarify?

Let's keep this as it is for now, and discuss Remotes vs staged dependencies in the next cycle. 

>   If we use it, I assume we need to add something that these are removed before sending the package to CRAN. Do you agree?

The `Remotes:` line is not allowed in a CRAN pkg. If it does appear, we need to remove it.

**Review by ddsjoberg — CHANGES_REQUESTED** [review]
Looks great! A couple of comments and responses.

File: vignettes/release_strategy.Rmd (hunk 2 above)
Comment (bundfussr) [reply vignettes/release_strategy.Rmd]: Thanks, I have combined the two steps into one.

File: vignettes/release_strategy.Rmd (hunk 1 above)
Comment (bundfussr) [reply vignettes/release_strategy.Rmd]: OK, I have combined this step and the following one. Does this make sense?

File: vignettes/release_strategy.Rmd (hunk 3 above)
Comment (bundfussr) [reply vignettes/release_strategy.Rmd:42]: I am not sure if we need the issue created by `use_release_issue()`. Looks like a repetition of the items we have here. I would prefer to keep it here (and not duplicate it in `release_bullets()` functions). Then the information is always available and maintained in one place.

If we need any additional checks, I would rather add them to our workflows than create new items in the list. However, I think our workflows are pretty good. In my experience, the code was ready for submission if our workflows had passed. Only sometimes some links were missed.

File: vignettes/release_strategy.Rmd [hunk 4]
```diff
@@ -39,34 +39,97 @@ updates via the community meetings, especially any anticipated breaking changes.
...
 1) If CRAN asks for modifications, repeat steps 1-2 as necessary.
-1) Once the package is accepted and available on CRAN, a [GitHub action](https://github.com/pharmaverse/admiral/actions/workflows/pages/pages-build-deployment) is set up to rebuild the `{admiral}` website with all the updates for this release.
-1) Use the release button on GitHub to "release" the package onto GitHub.  This release onto Github archives the version of code within the `main` branch, attaches the News/Changelog file, bundles the code into a `tar.gz` file and makes a validation report via the GitHub action `validation` from [insightsengineering/validatoR](https://github.com/insightsengineering/thevalidatoR).  Please see past [admiral releases](https://github.com/pharmaverse/admiral/releases) for reference and the [Releasing to Github](release_strategy.html#releasing-to-github) section for more details.
-
-**Quarterly Release:**  
-
-* `main >> patch` (To be prepared in case of a needed hotfix)
-
+1) Once the package is accepted and available on CRAN, use the release button on
+GitHub to "release" the package onto GitHub.  This releases onto Github archives
```
Comment (bms63) [inline vignettes/release_strategy.Rmd:59]: Should we link down to the "Releasing to GitHub" section as the tags is pretty important for the publishing part

File: vignettes/release_strategy.Rmd (hunk 3 above)
Comment (bms63) [reply vignettes/release_strategy.Rmd:42]: oohhh i like the release issue and the abilities to add additional items, e.g. the readme items for presentations, schedule that Edoardo has worked on.  

i already made one https://github.com/pharmaverse/admiral/issues/2256

Every release is pretty annoying honestly.  maybe some of these extra might help us.

Comment (ddsjoberg) [reply vignettes/release_strategy.Rmd:42]: I find the release issue very helpful even after having done many many releases to CRAN. 

But since it's Ben's burden, we should go with what's best for him as the maintainer.

Comment (bundfussr) [reply vignettes/release_strategy.Rmd:47]: Hm, maybe admiral is different. For admiralonco releasing was quite easy so far. I have added a paragraph mentioning `use_release_issue()`.

Comment (bms63) [reply vignettes/release_strategy.Rmd:47]: yeah every time there is some small issue from CRAN.  then the bugs! actually...can roche start checking the package now rather than after it gets to CRAN as you all's process surfaces some issues

Comment (bundfussr) [reply vignettes/release_strategy.Rmd:47]: I think checking admiral on company level before the release could avoid some patch release. We would need to change our release process for this (and most likely our branching strategy). At the moment we are updating the release candidate until a few days before the release. So we do not have enough time to update the company-specific code and test it with the new version. We would need to add a period between freezing the release candidate and submitting to CRAN.

Comment (ddsjoberg) [reply vignettes/release_strategy.Rmd:47]: it makes sense that admiralonco is an easier release because admiral has reverse dependencies, more users, and is a complex package in many ways.

File: vignettes/release_strategy.Rmd [hunk 5]
```diff
@@ -39,34 +39,95 @@ updates via the community meetings, especially any anticipated breaking changes.
...
+
+## Planned Release
+
+A package release is done in the following steps:
+
+1) Create the release version:
+    a) Create a new feature branch from `main`.
+    a) Call `usethis::use_version()` to update the `Versions:` field in the
+    `DESCRIPTION` file and update `NEWS.md` (replace "(development version)"
+    with the version to be released)
```
Comment (bms63) [inline vignettes/release_strategy.Rmd:55]: I think a note to double check that `Remotes` field is removed is needed here

File: vignettes/release_strategy.Rmd [hunk 6]
```diff
@@ -39,34 +39,95 @@ updates via the community meetings, especially any anticipated breaking changes.
...
+GitHub to "release" the package onto GitHub, select the `main` branch as target,
+and use a tag of the form `vX.Y.Z`, e.g., `v1.1.2` (see past [admiral
+releases](https://github.com/pharmaverse/admiral/releases) for reference and
+[Releasing to Github](release_strategy.html#releasing-to-github) section for
+more details).
+
+    This releases onto Github archives the version of code within the `main` branch,
+attaches the News/Changelog file, bundles the code into a `tar.gz` file and
+makes a validation report via the GitHub action `validation` from
+[insightsengineering/validatoR](https://github.com/insightsengineering/thevalidatoR).
```
Comment (bms63) [inline vignettes/release_strategy.Rmd:73]: Oh I think it is worth mentioning that we manually upload the R package `tar.gz` file as the other two contain all the files on the repo at the time of the release 
image

File: vignettes/release_strategy.Rmd [hunk 7]
```diff
@@ -39,34 +39,95 @@ updates via the community meetings, especially any anticipated breaking changes.
...
+
+    It also builds the website for the released version and stores it in a new
+`vx.y.z` folder on the `gh-pages` branch.
+
+1) Merge `main` into `patch` to be prepared in case of a needed hotfix.
+1) Create the next development version:
+    a) Create a new feature branch from `main`.
```
Comment (bms63) [inline vignettes/release_strategy.Rmd:80]: so I don't think we need to do this anymore.  We can create a patch branch off the tagged release.  See https://github.com/pharmaverse/admiral/pull/2298 for demo

**Comment by bms63** [comment]
Could #358 be linked somewhere in the discussion.  this was very handy in this last cycle

**Comment by bms63** [comment]
@zdz2101 can you help review this PR and incorporate any of learnings from the recent hotfix.  tyty!!

**Comment by bms63** [comment]
This little create branch from tag is nice to include - https://github.com/pharmaverse/admiraldev/issues/361

File: vignettes/release_strategy.Rmd [hunk 8]
```diff
@@ -39,34 +39,95 @@ updates via the community meetings, especially any anticipated breaking changes.
...
+branch, where bug fixes and new features would be released earlier than planned,
+thus avoiding the `patch` branch release.
+
+A hot fix release is done in the following steps:
+
+1) Identify all the bugs that need to be fixed for this hot fix release and
+label them with the "hotfix" label.
+1) Branches addressing the bugs should have Pull Requests merged into a single
+`patch` branch **NOT** the `main` branch, where the `patch` branch has been
+created from the most recent release of the package. There is no need to use
```
Comment (zdz2101) [inline vignettes/release_strategy.Rmd:99]:
```suggestion
created from the most recent release of the package. To do this, run the following git commands:

`git checkout -b patch vX.Y.Z`
`git push -u origin patch`

There is no need to use
```

**Review by zdz2101 — COMMENTED** [review]
Added a little blurb about how create the patch branch appropriately, otherwise I wish I read this earlier before I did it! It's perfect

**Review by zdz2101 — APPROVED** [review]
LGTM!

**Comment by bms63** [comment]
Thanks again @bundfussr and @ddsjoberg for helping get our Release Strategy in order.  Zelos and I made a few updates and this looks good to go!!
